reframe the governance

dcalvin opened the pull request:
Signed-off-by: Calvin Weng <wenghao@pingcap.com>

The current version is missing some important aspects of Governance. 
Based on the Governance for Vitess and Kubernetes, I updated the doc to make it more governance like.

winkyao commented on GOVERNACE.md:
```diff
@@ -1,45 +1,75 @@
+This is a Work in Progress. Stay tuned for more follow-up updates! 
```
Could you also modify https://github.com/tikv/community/blob/master/GOVERNACE-zh_CN.md in this PR?

dcalvin replied:
Will do later, after this one is submitted

winkyao commented on GOVERNACE.md:
```diff
@@ -1,45 +1,75 @@
+
+Each SIG must have a charter that specifies its scope (topics, code repos and directories), responsibilities, areas of authority, how members and roles of authority/leadership are selected/granted, how decisions are made, and how conflicts are resolved. See the SIG charter process for details on how charters are managed. SIGs should be relatively free to customize or change how they operate, with ...
+
+See [sig governance](../sig-governace/sig-governace.md) for more details about current SIG operating mechanics, such as mailing lists, meeting times, etc.
```
`../sig-governace/sig-governace.md` -> `/sig-governace/sig-governace.md`, absolute path is more appreciate

winkyao commented on GOVERNACE.md:
```diff
@@ -1,45 +1,92 @@
+The current list of PMC members are:
 
-Organizer Committee:  The Organizer Committee consists of organizers in charge of event or activity operations; in charge of the execution of PMC’s strategies and decisions. Specifically, the members are the leaders of user groups or community event organizers across regions
+- Max Liu(**[@goroutine](https://github.com/ngaut/)**)
```
I will propose the PMC List in this PR, https://github.com/tikv/community/pull/12 , and I create a new page to show the list. So I think there is no need to show the list here, otherwise, we need to maintain it in two places.

dcalvin replied:
OK~ I will add the link here instead.

winkyao commented:
@gingerkidney @Hoverbear PTAL

Hoverbear commented on GOVERNACE.md:
```diff
@@ -1,45 +1,87 @@
+
 # Governace
 
+The TiKV community is a meritocratic, consensus-based community project. Anyone with an interest in the project can join the community, contribute to the project design and participate in the decision making process. This document describes how that participation takes place and how to set about earning merit within the project community.
```
```suggestion
The TiKV community is a community project. Anyone with an interest in the project can join the community, contribute to the project design and participate in the decision making process. This document describes how that participation takes place and how to set about earning merit within the project community.
```

`Meritocratic` is not a welcoming word

Hoverbear commented on GOVERNACE.md:
```diff
@@ -1,45 +1,87 @@
+The project is comprised of the following types of subgroups:
+
+- Project Management Committee, PMC
+- Organizer Committee
```
Maintainers?

dcalvin replied:
Maintainer is a role under Developer Group, and is detailed in Membership.  Here we talk about the subgroup levels, which include the listed groups. Does it make sense to you?

Hoverbear commented:
I'm very confused, I don't see `maintainers` at all.

Soline324 commented on GOVERNACE.md:
```diff
@@ -1,45 +1,87 @@
 
-* SIG: Special Interest Group, need invitation currently. Members are Contributors who are active in a particular project. The ultimate goal of the group is to train community talents and help them advance to the higher-level Reviewer, Committer and Maintainer.
-* WG:  Working group open for everyone, community members can join without a threshold. The working group is a temporary group established to complete a short-term project, and the project is dissolved when it is completed. A working group can involved in multiple interest groups. After a period of training, excellent community members will be selected into a special interest group for further st ...
+Decisions about the future of the project are made by the PMC. New proposals and ideas (changes of major features, organization, or processes) can be brought to the PMC’s attention through the [Request for Change (RFC)](https://github.com/tikv/rfcs)process. For the change to happen, the RFC must earn the majority votes in the PMC.
```
```suggestion
Decisions about the future of the project are made by the PMC. New proposals and ideas (changes of major features, organization, or processes) can be brought to the PMC’s attention through the [Request for Comments (RFC)](https://github.com/tikv/rfcs)process. For the change to happen, the RFC must earn the majority votes in the PMC.
```

dcalvin commented:
> I'm very confused, I don't see `maintainers` at all.

Added.

dcalvin commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,94 @@
+
+### Maintainers
+
+While the PMC is the core management body that oversees the project, Maintainers are the technical authority in the Developer Group who function as planners and designers of the project. Maintainers set technical direction and priorities for the sub-project, and ensure its continued health and development. They make or approve design decisions for the project - either directly or by delegating th ...
```
Added Maintainers here @Hoverbear

dcalvin commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,94 @@
+
+Membership of the PMC is by invitation from the existing PMC members. A nomination will result in discussion and then a vote by the existing PMC members. PMC membership votes are subject to consensus approval of the current PMC members.
+
+Given that PingCAP has built and supported this project in the first place, they reserve the right to appoint and form the first committee for a period of first year from the time this document becomes effective. Future PMC numbers are generated based on the rules described above.
```
Some explanation regarding the current PMC members and how future PMC is generated

Hoverbear replied:
I don't think the general tone of the first sentence is really in line with our communication. Let's pick something more humble:

> For the first formation of the committee our maintainers nominated a group of people who had great impact or influence on the project, including some members from PingCAP who founded TiKV. Future PMC numbers are generated based on the rules described above.

dcalvin replied:
Sounds great!

BusyJay commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,94 @@
+
+### Maintainers
+
+While the PMC is the core management body that oversees the project, Maintainers are the technical authority in the Developer Group who function as planners and designers of the project. Maintainers set technical direction and priorities for the sub-project, and ensure its continued health and development. They make or approve design decisions for the project - either directly or by delegating th ...
```
> ...core management body that oversees...

The word "body" looks weird to me. Maybe person is better?

dcalvin replied:
It's not a person, more of a group of people.

BusyJay replied:
"body" seems more like a biological word to me. Can you show me similar usages?

Hoverbear replied:
"A statutory authority is a body set up by law which is authorised to enact legislation on behalf of the relevant country or state." - https://en.wikipedia.org/wiki/Statutory_authority

BusyJay commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,94 @@
+
+## Decision Making Process
+
+Decisions about the future of the project are made by the PMC. New proposals and ideas (changes of major features, organization, or processes) can be brought to the PMC’s attention through the [Request for Change (RFC)](https://github.com/tikv/rfcs)process. For the change to happen, the RFC must earn the majority votes in the PMC.
```
So although maintainers "set technical direction and priorities for the sub-project" and "make or approve design decisions for the project", but a features proposal still needs majority votes to get it merged?

dcalvin replied:
Agree.

dcalvin replied:
Maybe need further clarification here.  What about now?

BusyJay commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,94 @@
+
+Given that PingCAP has built and supported this project in the first place, they reserve the right to appoint and form the first committee for a period of first year from the time this document becomes effective. Future PMC numbers are generated based on the rules described above.
+
+The number of PMC members should be limited to 6. This number is chosen to ensure that sufficient points of view are represented, while preserving the efficiency of the decision making process. For a list of the current PMC members, see the [PMC Members](/committee/.README.md#members).
```
Why not choose an odd number?

dcalvin replied:
Remove this number for now for future decision.

Hoverbear commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,94 @@
+
+The TiKV community follows the TiKV Code of Conduct. Here are some excerpts:
+
+> In the interest of fostering an open and welcoming environment, we as contributors and maintainers pledge to making participation in our project and our community a harassment-free experience for everyone, regardless of age, body size, disability, ethnicity, sex characteristics, gender identity and expression, level of experience, education, socio-economic status, nationality, personal appearan ...
```
Link to our existing project wide CoC after please

Hoverbear commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,94 @@
+
+## Decision Making Process
+
+Depending on the nature, decisions about the future of the project are made by the PMC or by Maintainers. As the technical authority, Maintainers set technical direction and priorities for the sub-project, while PMC make other high-level decisions like establishment and operation policies of sub-projects, sub-structures, promoting new Maintainers, security processes, etc. Tech New proposals and i ...
```
```suggestion
Depending on the nature, decisions about the future of the project are made by the PMC or by Maintainers. As the technical authority, Maintainers set technical direction and priorities for the sub-project, while PMC make other high-level decisions like establishment and operation policies of sub-projects, sub-structures, promoting new Maintainers, security processes, etc. Tech New proposals and ideas (changes of major features, organization, or processes) can be brought to the PMC or Maintainers’ attention through the [Request for Comments (RFC)](https://github.com/tikv/rfcs) process. For the change to happen, the RFC must earn the majority votes in the corresponding group.
```

winkyao commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,94 @@
+
+### Maintainers
+
+While the PMC is the core management group that oversees the project, Maintainers are the technical authority in the Developer Group who function as planners and designers of the project. Maintainers set technical direction and priorities for the sub-project, and ensure its continued health and development. They make or approve design decisions for the project - either directly or by delegating t ...
```
with anonymous votes?

dcalvin replied:
changed to majority

Hoverbear commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,94 @@
+
+Membership of the PMC is by invitation from the existing PMC members. A nomination will result in discussion and then a vote by the existing PMC members. PMC membership votes are subject to consensus approval of the current PMC members.
+
+For the first formation of the committee our maintainers nominated a group of people who had great impact or influence on the project, including some members from PingCAP who founded TiKV. Future PMC numbers are generated based on the rules described above.
```
```suggestion
For the first formation of the committee our maintainers nominated a group of people who had great impact or influence on the project, including some members from PingCAP who founded TiKV. New PMC members are selected based on the rules described above.
```

gingerkidney commented on GOVERNANCE.md:
```diff
@@ -0,0 +1,94 @@
+
+Members of the PMC do not have significant authority over other members of the community, although it is the PMC that votes on new Maintainers or Committers, and makes all major decisions for the future with respect to TiKV, such as project-level governance policies, management of sub-structures, security processes and so on. It also makes decisions when community consensus cannot be reached. In  ...
+
+Membership of the PMC is by invitation from the existing PMC members. A nomination will result in discussion and then a vote by the existing PMC members. PMC membership votes are subject to consensus approval of the current PMC members.
```
Should we have a PMC nominated criteria and vote procedure?

dcalvin replied:
Yes. That's to be updated.

dcalvin commented:
Sorry that the diff looks a bit messy. Basically in this commit, I:

- Trimmed some info in PMC
- Added the Principles section
- Added Decision making and voting
- Added  Conflict resolution
- Added Adding new projects to the TiKV GitHub organization
- Added  Changes in Governance

@Hoverbear @winkyao @gingerkidney @siddontang @BusyJay  PTAL again. Thanks

Hoverbear commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,120 @@
+
+# TiKV Governance
+
+This document describes the governance rules of the TiKV project (organization). It is meant to be followed by all the repositories in the project and the TikV community.
```
```suggestion
This document describes the governance rules of the TiKV project (organization). It is meant to be followed by all the repositories in the project and TiKV contributors.
```

Hoverbear commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,120 @@
+### Developer Group
 
-## Developer Group Structure
+As the cornerstone of community development, the TiKV Developer Group consists of these roles: Maintainer, Committer, Reviewer, Active Contributor, and Contributor. Each role takes corresponding responsibilities in the community. Collectively they play an important role in the robust development of TiDB. For more details, refer to [Community Membership](/community-membership.md).
```
```suggestion
As the cornerstone of community development, the TiKV Developer Group consists of these roles: Maintainer, Committer, Reviewer, Active Contributor, and Contributor. Each role takes corresponding responsibilities in the community. Collectively they play an important role in the robust development of TiKV. For more details, refer to [Community Membership](/community-membership.md).
```

Hoverbear commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,120 @@
+
+## Decision making and voting
+
+Decisions are build on consensus between PMC members or Maintainers. Depending on the nature, decisions about the future of the project are made by the PMC or by Maintainers. As the technical authority, Maintainers set technical direction and priorities for the sub-project, while PMC make other high-level decisions like establishment and operation policies of sub-projects, sub-structures,promotin ...
```
```suggestion
Decisions are build on consensus between PMC members or Maintainers. Depending on the nature, decisions about the future of the project are made by the PMC or by Maintainers. As the technical authority, Maintainers set technical direction and priorities for the sub-project, while PMC make other high-level decisions like establishment and operation policies of sub-projects, sub-structures, promoting new Maintainers, security processes, etc.
```

Hoverbear commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,120 @@
+
+## Decision making and voting
+
+Decisions are build on consensus between PMC members or Maintainers. Depending on the nature, decisions about the future of the project are made by the PMC or by Maintainers. As the technical authority, Maintainers set technical direction and priorities for the sub-project, while PMC make other high-level decisions like establishment and operation policies of sub-projects, sub-structures,promotin ...
```
Seems like maintainers should guide/choose our technical priorities?

dcalvin replied:
> There aren't sub project maintainers defined.

Do you mean we need to define our sub-projects clearly?

Hoverbear commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,120 @@
+
+Decisions are build on consensus between PMC members or Maintainers. Depending on the nature, decisions about the future of the project are made by the PMC or by Maintainers. As the technical authority, Maintainers set technical direction and priorities for the sub-project, while PMC make other high-level decisions like establishment and operation policies of sub-projects, sub-structures,promotin ...
+
+New proposals and ideas (changes of major features, organization, or processes) can be brought to the PMC or Maintainers’ attention through the [Request for Comments (RFC)](https://github.com/tikv/rfcs) process. For the change to happen, the RFC must earn the supermajority (2/3) votes in the corresponding group.
```
This includes **all** major changes, those the PMC or maintainers are wanting to enact, or those our contributors want to push, yes?

So all new `Feature` tagged PRs to any TiKV owned project needs an RFC?

dcalvin replied:
Isn't it our current practice to submit RFC for any new features to be developed?

Hoverbear replied:
I want to make sure this is a "No exceptions" rule. While technically it's our practice we don't follow it religiously.

dcalvin replied:
Yeah I get what you mean. It's also my intention to state things as clear as possible so that there is less room not to follow them

Hoverbear commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,120 @@
+
+## Conflict resolution
+
+In general, we prefer that technical issues and maintainer membership are amicably worked out between the persons involved. If a dispute cannot be decided independently, depending on the nature, the PMC or Maintainers can be called in to resolve the issue by voting. The same PR can be used or a separate PR can be opened in the concerned repository for voting.
```
Maintainer membership is already defined clearly above, not sure what is to be "worked out between persons"

dcalvin replied:
Maybe just technical issues...

Hoverbear replied:
Perhaps we should also note the CoC here.

Hoverbear commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,120 @@
+
+## Adding new projects to the TiKV GitHub organization
+
+New projects can be added to the TiKV organization via GitHub issue discussion in one of the existing projects in the organization, as long as as they adheres to the [CNCF charter](https://www.cncf.io/about/charter/) and the guidelines in this governance. Once sufficient discussions have taken place (normally 3-5 business days but also depending on the volume of conversation), the Maintainers of  ...
```
```suggestion
New projects can be added to the TiKV organization via GitHub issue discussion in one of the existing projects in the organization, as long as as they adheres to the [CNCF charter](https://www.cncf.io/about/charter/) and the guidelines in this governance. Once sufficient discussions have taken place, the Maintainers of the project where the issue was opened (since different projects in the organization may have different maintainers) will decide whether the new project should be added. The requester also needs to create an corresponding RFC for the change to happen, as described in [Decision Making and Voting](#decision-making-and-voting).
```

There is no precedent for this, so we can't claim that.

Hoverbear commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,124 @@
+The PMC functions as the core management team that oversees the TiKV community. The PMC has additional responsibilities over and above those of Maintainers. These responsibilities ensure the smooth running of the project.
 
-Developer Group:
+Members of the PMC do not have significant authority over other members of the community, although it is the PMC that votes on new Maintainers or Committers, and makes all major decisions for the future with respect to TiKV, such as project-level governance policies,management of sub-structures, security processes and so on. It also makes decisions when community consensus cannot be reached. In a ...
```
```suggestion
Members of the PMC do not have significant authority over other members of the community, although it is the PMC that votes on new Maintainers or Committers, and makes all major decisions for the future with respect to TiKV, such as project-level governance policies, management of sub-structures, security processes and so on. It also makes decisions when community consensus cannot be reached. In addition, the PMC has access to the project’s private mailing list and its archives.
```

Hoverbear replied:
We should note that the PMC executed the ban/exile folks who violate the Code of Conduct, or in special cases (such as sabotage) as found by a supermajority of the PMC.

Hoverbear commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,124 @@
-## Developer Group Structure
+The current number of PMC members is limited to 7. This is to ensure both sufficient representation of different groups in TiKV community and efficiency of decision making.
+
+For a list of the current PMC members, see the [PMC Members](/committee/.README.md#members).
```
Is this the correct fiile path? It's `.README.md`?

Hoverbear commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,124 @@
+- Organizer Committee
+- Developer Group
+    - Maintainers
+    - Special Interest Groups, SIGs (Tech Leads)
```
```suggestion
    - Special Interest Groups, SIGs
```

Hoverbear commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,124 @@
+- Developer Group
+    - Maintainers
+    - Special Interest Groups, SIGs (Tech Leads)
+    - Working Groups, WGs (Organizer)
```
```suggestion
    - Working Groups, WGs
```

Hoverbear commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,124 @@
+### Project Management Committee
 
-PMC: The Project Management Committee; functions as the core management team that oversees the TiDB community. The PMC decides the roadmap of the project and makes major decisions related to the community.
+The PMC functions as the core management team that oversees the TiKV community. The PMC has additional responsibilities over and above those of Maintainers. These responsibilities ensure the smooth running of the project.
```
```suggestion
The PMC functions as the core management team that oversees the TiKV community. While maintainers guide the project on technical matters, the PMC is responsible for stewarding, promoting, and fostering the project. These responsibilities include ensuring the smooth running and continued growth of the project.
```

nrc commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,107 @@
+
+Proposals and ideas can be submitted for agreement via a github issue or PR. Depending on the nature, decisions about the future of the project are made by the PMC or by Maintainers. As the technical authority, Maintainers set technical direction and priorities for the SIG projects, while the PMC makes other high-level decisions like establishment and operation policies of sub-projects, sub-struc ...
+
+Major changes such as feature proposals and organization or process changes should be brought to the PMC or Maintainers’ attention through the [Request for Comments (RFC)](https://github.com/tikv/rfcs) process. For the change to happen, the RFC must earn the supermajority (2/3) votes in the corresponding group.
```
Again, I would say RFCs are accepted by consensus, not by vote. I think it would be good to let SIGs approve RFCs which will only affect their area and the PMC has to approve cross-cutting or strategic RFCs.

nrc commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,107 @@
+
+## Adding new projects
+
+New projects can be added to the TiKV organization via GitHub issue discussion in one of the existing SIG, as long as as they adheres to the [CNCF charter](https://www.cncf.io/about/charter/) and the guidelines in this governance. Once sufficient discussions have taken place, the Maintainers will decide whether the new project should be added. The requester needs to create an corresponding RFC fo ...
```
"in one of the existing SIG" what does it mean?

dcalvin replied:
updated

nrc commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,107 @@
 
-- Tech Leader: The technical leader of the group controls the technical direction of the entire team. Responsible for formulating study plans, member assessment rules, etc. A team can have multiple Tech Leaders.
-- Please see the community organization chart for the rest of the roles
+The contents of this document are based on <http://oss-watch.ac.uk/resources/meritocraticgovernancemodel> by Ross Gardler and Gabriel Hanganu, and [Kubernetes Governance](https://github.com/kubernetes/community/blob/master/governance.md).
```
I would suggest that basing the governance model on Kubernates is inappropriate because it is a well-established, large project, with  many large orgs involved who are materially affected by K8s decisions. TiKV should look to smaller projects for governance models (e.g., Ember, Rust, Linkerd, etc) and choose a more lightweight model.

nrc commented on README.md:
```diff
@@ -8,9 +8,9 @@ Your journey of becoming a contributor and committer starts from here: improving
-+ Working Groups are temporary groups that are formed to address issues that cross SIG boundaries. Working groups do not own any code or other long term artifacts. Working groups can report back and act through involved SIGs. See more details in WG governance([中文](/committee/wg-governance/wg-governance-zh_CN.md), [English](committee/wg-governance/wg-governance.md))
++ PMC (Project Management Committee) is to manage and oversee the entire TiKV community.
++ Special Interest Groups (SIGs) are persistent open groups that focus on a part of the project. SIGs must have open and transparent proceedings. Anyone is welcome to participate and contribute provided they follow the [TiKV Code of Conduct](https://github.com/tikv/tikv/blob/master/CODE_OF_CONDUCT.md). The purpose of a SIG is to own and develop a set of TiKV subprojects or components. See more de ...
++ Working Groups are temporary groups that are formed to address issues that cross SIG boundaries. Working groups do not own any code or other long term artifacts. Working groups can report back and act through involved SIGs. 
```
This is all just duplication of governance.md, I would remove it and leave the link

dcalvin commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,107 @@
 
-Community groups include Special Interest Group (SIG) and Working Group (WG). The second part of the organizational structure illustrates the definition of the SIG and WG, the internal roles of the group, and the role promotion machanism.
+### Maintainers
+
```
I think Maintainers are still the core group here, as TiKV is pretty hardcore in the technical sense. This is so that Maintainers can be focus on the tech aspect of the work and let PMC handle other governance work.

dcalvin commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,107 @@
+
+#### SIGs
+
+The TiKV project is organized primarily into Special Interest Groups, or SIGs. Each SIG consists of members from multiple organizations or individuals, with a common purpose of advancing the TiKV project for a specific topic, such as Coprocessor or Documentation. The goal of an SIG is to enable a distributed decision structure and code ownership, as well as providing focused forums for getting wo ...
```
Good suggestion. @zhangjinpeng1987

dcalvin commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,107 @@
+
+The TiKV project is organized primarily into Special Interest Groups, or SIGs. Each SIG consists of members from multiple organizations or individuals, with a common purpose of advancing the TiKV project for a specific topic, such as Coprocessor or Documentation. The goal of an SIG is to enable a distributed decision structure and code ownership, as well as providing focused forums for getting wo ...
+
+Currently, any Contributor within a SIG is immediately eligible member of that SIG. It is our intention to properly guide or mentor community members, and help them advance to the higher-level Reviewer, Committer within the SIG.
```
Contributor becomes a SIG member immediately, but that SIG has to exist

dcalvin commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,107 @@
+
+Each SIG must have a charter that specifies its scope (topics, code repositories, and directories), responsibilities, areas of authority, how members and roles of authority/leadership are selected/granted, how decisions are made, and how conflicts are resolved. See the [SIG charter template](/sig-governance/SIG-CHARTER-TEMPLATE.md) for details on how charters are formed and managed. SIGs should b ...
+
+A primary reason that SIGs exist is as forums for collaboration. Much work in a SIG should stay local within that SIG. However, SIGs must communicate to ensure other SIGs and community members can find notes of meetings, discussions, designs, and decisions, and periodically communicate a high-level summary of the SIG's work to the community.
```
Like SIG meetings biweekly or monthly?

dcalvin commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,107 @@
+
+A primary reason that SIGs exist is as forums for collaboration. Much work in a SIG should stay local within that SIG. However, SIGs must communicate to ensure other SIGs and community members can find notes of meetings, discussions, designs, and decisions, and periodically communicate a high-level summary of the SIG's work to the community.
+
+SIGs are organized and operated by Tech Leads, which is a SIG internal role that oversees the health and sustained development of the SIG. Upon the initial establishment of a SIG, the maintainers will assign 2-3 Tech Leads.
```
Or SIG Lead?

dcalvin commented:
Based on the [TiKV Maintainer Meeting on Governance](https://docs.google.com/document/d/1czn_BAEaymLDjZ1vb7UeUjTm1WiDV5jcUoNnCeJojfc/edit?ts=5e761a2d)(In Chinese), I refactored the governance with the following points incorporated:
-  Adjusted the maintainer promotion role to allow key user/ecosystem contributors to have a more practical path 
- Remove PMC from the structure for more simplified and clearer governance
-  Incorporate membership in governance to remove information redundancy
- Remove WG, for now, to allow more efficient governance
- Update consensus-based decision making and specify voting scenarios
- Other necessary updates as necessary   @winkyao @nrc @sunxiaoguang @shenli @siddontang @zhangjinpeng1987 @BusyJay PTAL

dcalvin commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,107 @@
-The community structure comprises two parts. The first part is the community organization, and the second part is the role division within the community group and the promotion path of each community role.
+The project is comprised of and operated by the following subgroups:
+
+- Project Management Committee (PMC)
```
Removed OMC

BusyJay commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,117 @@
+| -----| ---------------- | ------------ | ------- |
+Contributor | contribute | initial contribution (PR, issue reporting, answering questions, etc.) | `Contributor List`|
+| Active Contributor | active contributor in the community | sponsored by 2 Reviewers. Multiple contributions (8+) to the project. | `SIG member`  of the SIG |
+| Reviewer | review code contributions | Active Contributor + History of review and authorship within a SIG | `SIG member` of the SIG |
```
`+` seems informal.

BusyJay commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,117 @@
+- Other cases where consensus cannot be made (super majority)
 
-## Developer Group Structure
+For formal votes, a specific statement of what is being voted on should be added to the relevant github issue or PR, and a link to that issue or PR added to the maintainers meeting agenda document. Maintainers should indicate their yes/no vote on that issue or PR, and after a suitable period of time, the votes will be tallied and the outcome noted.
```
There may be some circumstance that needs secret ballot, though I can't think of any for now. But I'm OK to add solve it when we really need to.

nrc commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,117 @@
+
+# TiKV Governance
+
+ This document describes the governance rules of the TiKV project (organization). It is meant to be followed by all the repositories in the project and the TiKV project.
```
```suggestion
 This document describes the governance rules of the TiKV project (organization). It is meant to be followed by all the repositories in the TiKV organization.
```

nrc commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,117 @@
+
+### Maintainer
+
+Maintainers are first and foremost contributors that have shown they are committed to the long term success of a project. They are the planners and designers of the TiKV project, with the authority to merge branches into the master branch. Maintainership is about building trust with the current maintainers of the project and being a person that they can depend on to make decisions in the best int ...
```
"authority to merge branches into the master branch" - this is a confusing statement - we don't merge release branches and technically we don't merge PRs - the bot does, but also any reviewer can approve the bot to merge, not just maintainers.

dcalvin replied:
Will  remove this.

nrc commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,117 @@
+
+Maintainers are first and foremost contributors that have shown they are committed to the long term success of a project. They are the planners and designers of the TiKV project, with the authority to merge branches into the master branch. Maintainership is about building trust with the current maintainers of the project and being a person that they can depend on to make decisions in the best int ...
+
+While maintainership indicates a valued member of the community who has demonstrated a healthy respect for the project’s aims and objectives, their work must be reviewed by the community before acceptance in an official release.
```
```suggestion
While maintainership indicates a valued member of the community who has demonstrated a healthy respect for the project’s aims and objectives, their work must still be reviewed by the community before being accepted into an official release.
```

nrc commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,117 @@
+
+A Maintainer is not allowed to merge their change without approval from another person. However, they are allowed to sidestep this rule under justifiable circumstances. For example:
+
+- If a CI tool is broken, they may override the tool to still submit the change.
```
The change should still be reviewed it is just *merged* manually, so this does not quite fit.

nrc commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,117 @@
+A Maintainer is not allowed to merge their change without approval from another person. However, they are allowed to sidestep this rule under justifiable circumstances. For example:
+
+- If a CI tool is broken, they may override the tool to still submit the change.
+- Minor typos or fixes for broken tests.
```
I think these should still get review - if they are very minor, then they are very quick to review

nrc commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,117 @@
+
+- If a CI tool is broken, they may override the tool to still submit the change.
+- Minor typos or fixes for broken tests.
+- The change was approved through other means than the standard process.
```
Approved by who? I think what is implied by this is that if a PR is reviewed offline, then a maintainer is free to use that review without a GitHub review (e.g, they can just write something like "reviewed by @nrc offline" in the PR and make the bot merge it).

I think emergency merging should be kept to absolute emergencies - e.g., ongoing security breach, accidentally exposed password, etc. I think it is very easy to have a relaxed attitude to "justifiable circumstances" and unintentionally cause problems.

nrc commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,117 @@
+Contributors wanting to become maintainers are expected to:
+
+- Enable and promote successful adoptions on the user or ecosystem end
+- Actively Propose features to TiKV that are later implemented with great significance to the project direction and the eco-system
```
I think more important than proposing is the ongoing development - that might be all in RFCs, not code, or might be both. An antipattern is that contributor X proposes a feature without enough work, then contributor Y takes over and comes up with a working design and contributors Y and Z implement it. Just because X was the first to suggest it, doesn't mean they are a great contributor.

dcalvin replied:
So you mean the actual implementation and development matter more than just RFCs?

nrc replied:
Not exactly. Developing an RFC might be important. Writing a *good* RFC is important (whether it is implemented or not). But writing a bad RFC is not important, even if somebody else develops it into a good RFC or implements it (in which case the latter two contributions are important, but the initial (bad) RFC is not).

nrc commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,117 @@
+
+- Enable and promote successful adoptions on the user or ecosystem end
+- Actively Propose features to TiKV that are later implemented with great significance to the project direction and the eco-system
+- Able to promote substained collaboration within the project
```
```suggestion
- Able to promote sustained collaboration within the project
```

I'm not sure what this is getting at? Do we want the candidate to have sustained contribution? Or to encourage others to collaborate or to stick around? (This last one is very hard in an open source environment and I don't think we should expect maintainers to be able to do it).

dcalvin replied:
OK. Removing this

nrc commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,117 @@
+- Enable and promote successful adoptions on the user or ecosystem end
+- Actively Propose features to TiKV that are later implemented with great significance to the project direction and the eco-system
+- Able to promote substained collaboration within the project
+- Demonstrate a deep and comprehensive understanding of TiKV's codebase, technical goals, and directions
```
```suggestion
- Demonstrate a deep and comprehensive understanding of TiKV's architecture, technical goals, and directions
```

A maintainer who contributes a lot to the project without coding (e.g., docs, testing, or organizational) might not know the codebase deeply, but should understand how TiKV works at a high level.

dcalvin replied:
valid point.

nrc commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,117 @@
+- Actively Propose features to TiKV that are later implemented with great significance to the project direction and the eco-system
+- Able to promote substained collaboration within the project
+- Demonstrate a deep and comprehensive understanding of TiKV's codebase, technical goals, and directions
+- Handle complex problems in the code implementation process
```
Again, I think maintainer candidates might should be eligible without coding - they might have other skills which are valuable to the project.

nrc commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,117 @@
+- Demonstrate a deep and comprehensive understanding of TiKV's codebase, technical goals, and directions
+- Handle complex problems in the code implementation process
+
+A new Maintainer must be self-nominated or nominated by an existing Maintainer by updating [TiKV Maintainers](https://github.com/tikv/tikv/blob/master/MAINTAINERS.md#the-tikv-maintainers). This triggers a voting process that that requires supermajority votes from the current Maintainers.
```
```suggestion
A new Maintainer must be nominated by an existing Maintainer by updating [TiKV Maintainers](https://github.com/tikv/tikv/blob/master/MAINTAINERS.md#the-tikv-maintainers). This triggers a voting process that that requires supermajority votes from the current Maintainers.
```

I would suggest consensus rather than voting - if somebody can't be persuaded somebody will be a good maintainer then that is a very bad sign.

winkyao replied:
```suggestion
A new Maintainer must be self-nominated or nominated by an existing Maintainer by updating [TiKV Maintainers](https://github.com/tikv/tikv/blob/master/MAINTAINERS.md#the-tikv-maintainers). This triggers a voting process that requires supermajority votes from the current Maintainers.
```

nrc commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,117 @@
+A new Maintainer must be self-nominated or nominated by an existing Maintainer by updating [TiKV Maintainers](https://github.com/tikv/tikv/blob/master/MAINTAINERS.md#the-tikv-maintainers). This triggers a voting process that that requires supermajority votes from the current Maintainers.
+
+If a Maintainer is no longer interested or cannot perform the Maintainer duties listed above, they should volunteer to be moved to emeritus status. In extreme cases this can also occur by a vote of the maintainers per the voting process below.
+
```
Again, I think consensus of active maintainers is better than voting.

winkyao commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,117 @@
+
+### Maintainer
+
+Maintainers are first and foremost contributors that have shown they are committed to the long term success of a project. They are the planners and designers of the TiKV project, with the authority to merge branches into the master branch. Maintainership is about building trust with the current maintainers of the project and being a person that they can depend on to make decisions in the best int ...
```
```suggestion
Maintainers are first and foremost contributors that have shown they are committed to the long term success of a project. They are the planners and designers of the TiKV project. Maintainership is about building trust with the current maintainers of the project and being a person that they can depend on to make decisions in the best interest of the project in a consistent manner.
```

nrc commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,117 @@
+
+The TiKV project is organized primarily into Special Interest Groups, or SIGs. Each SIG consists of contributors with a common purpose of advancing the TiKV project for a specific topic, such as Coprocessor, Transaction, or Scheduling. The goal of an SIG is to enable a distributed decision structure and code ownership, as well as providing focused forums for getting work done, making decisions, a ...
+
+Each SIG must have a charter that specifies its scope (topics, code repositories, and directories), responsibilities, areas of authority, how members and roles of authority/leadership are selected/granted, etc. See the [SIG charter template](/sig-governance/SIG-CHARTER-TEMPLATE.md) for details on how charters are formed and managed. SIGs should be relatively free to customize or change how they o ...
```
```suggestion
Each SIG must have a charter that specifies its scope (topics, code repositories, and directories), responsibilities, areas of authority, how members and roles of authority/leadership are selected/granted, etc. See the [SIG charter template](/sig-governance/SIG-CHARTER-TEMPLATE.md) for details on how charters are formed and managed. SIGs should be relatively free to customize or change how they operate, within some broad guidelines and constraints imposed by this document and [Sig Governance](/committee/sig-governance/sig-governance.md).
```

dcalvin replied:
Anything changed here?

nrc replied:
I think there was a typo which was already fixed.

nrc commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,117 @@
+
+#### SIG Roles
+
+Within a SIG, you could find your path of contribution and growth through multiple roles - Contributor, Active Contributor, Reviewer, and Committer, each with corresponding responsiblities and requirements, as listed below:
```
Could we merge reviewer and committer? I think there is not much difference between them

dcalvin replied:
There are some consistent settings with the TiDB community side, based on which they are building  this contributor credit system. Plus,  the difference between Committer is they have different  privileges. Committer is the highest level within a SIG. Let's just leave this as is for now.

nrc commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,117 @@
+The project asks its users to participate in the project and community as much as possible. User contributions enable the project team to ensure that they are satisfying the needs of those users. Common user contributions include (but are not limited to):
 
-## Community Structure
+- evangelising about the project (e.g. a link on a website and word-of-mouth awareness raising)
```
```suggestion
- evangelizing the project (e.g. a link on a website and word-of-mouth awareness raising)
```

nrc commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,117 @@
 
-## Community Structure
+- evangelising about the project (e.g. a link on a website and word-of-mouth awareness raising)
+- informing developers of strengths and weaknesses from a new user perspective
```
```suggestion
- informing developers of strengths and weaknesses from a user's perspective
```

nrc commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,117 @@
-## Community Structure
+- evangelising about the project (e.g. a link on a website and word-of-mouth awareness raising)
+- informing developers of strengths and weaknesses from a new user perspective
+- providing moral support
```
"moral support" is a bit colloquial/informal

nrc commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,117 @@
+- evangelising about the project (e.g. a link on a website and word-of-mouth awareness raising)
+- informing developers of strengths and weaknesses from a new user perspective
+- providing moral support
+- providing financial support
```
to whom? How?

nrc commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,117 @@
+## Approving PRs
 
-PMC: The Project Management Committee; functions as the core management team that oversees the TiDB community. The PMC decides the roadmap of the project and makes major decisions related to the community.
+Unless otherwise requested by Maintainers, PRs may be merged after receiving at least two approvals from Reviewers or higer roles.
```
```suggestion
PRs may be merged after receiving at least two approvals from Reviewers (or more privileged) in the relevant SIGs.
```

nrc commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,117 @@
-- Reviewer: Reviewers are responsible for reviewing the code submitted by the community to ensure the quality and correctness; generated out of Contributors. Reviewer has the rights to approve code reviews with LGTM (Looks Good To Me).
-- Active contributor: Contributed 8 PRs and above to TiKV within one year and received recognition from 2 incumbent Reviewers
-- Contributor: Merged at least 1 PR in any Repo under [TiKV](https://github.com/tikv) 
+Ideally, all project decisions are resolved by consensus via a PR or GitHub issue. If this is not possible, Maintainers may call a vote. Many of the day-to-day project maintenance can be done by a [lazy consensus](http://communitymgt.wikia.com/wiki/Lazy_consensus) model. But the following items must be called to vote:
```
I think most decisions can be discussed on GH, but some cannot (e.g., whether to ban a member of the community). I think the discussion venue (GH, private meeting, etc) is orthogonal to the decision making mechanism (voting, consensus, etc).

dcalvin replied:
So you are suggesting we add something like a private maintainer meeting for

nrc replied:
I don't think we need to put it here. If the maintainers want an ad hoc or a regular meeting, then they can schedule one.

nrc commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,117 @@
+- Changing the governance rules in this document (super majority)
+- Licensing and intellectual property changes (including new logos, wordmarks) (simple majority)
+- Adding, archiving, or removing SIG/subproject (simple majority)
+- Other cases where consensus cannot be made (super majority)
```
These are important things, but having a vote makes them easier to happen than reaching consensus, which I think is the opposite of what is desired.

dcalvin replied:
What do you suggest?

nrc replied:
Consensus for everything :-)

dcalvin replied:
So what are occasions for voting？

nrc commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,117 @@
+### Proposal process
 
-### Organization
+We use a [Request for Comments (RFC) process for any substantial changes to TiKV. This process involves an upfront design that will provide increased visibility to the community. If you're considering a PR that will bring in a new feature that may affect how TiKV is implemented, or may be a breaking change, then you should start with a RFC. We've got the process documented in [RFC repository](htt ...
```
```suggestion
We use a Request for Comments (RFC) process for any substantial changes to TiKV. This process involves some upfront design that will provide increased visibility to the community. If you're considering a PR that will bring in a new feature that may affect how TiKV is implemented, or may be a breaking change, then you should start with a RFC. The process is documented in [RFC repository](https://github.com/tikv/rfcs) and have a [template](https://github.com/tikv/rfcs/blob/master/template.md) for you to get started.
```

I think for all but the most obvious RFCs, the best way to start is by discussing with the relevant SIG in an informal way, then write the RFC, perhaps with help from someone in the SIG if there is interest. This avoids wasted work and disappointment

dcalvin replied:
Sounds good

nrc commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,117 @@
+## Adding new SIGs/projects
 
-### Internal role:
+New SIGs or sub-projects can be added to the TiKV organization via [RFC process](#proposal-process), as long as as they adhere to the [CNCF charter](https://www.cncf.io/about/charter/) and the guidelines in this governance. Once sufficient discussions have taken place under the RFC PR, the Maintainers will call a vote to decide whether the new SIG/project should be added.
```
```suggestion
New SIGs or sub-projects can be added to the TiKV organization via [RFC process](#proposal-process), as long as as they adhere to the [CNCF charter](https://www.cncf.io/about/charter/) and the guidelines in this governance. Once sufficient discussions have taken place under the RFC PR, the Maintainers will call a vote to decide whether the new SIG/project should be added.
```
```suggestion
New SIGs or sub-projects can be added to the TiKV organization via [RFC process](#proposal-process), as long as as they adhere to the [CNCF charter](https://www.cncf.io/about/charter/) and the guidelines in this document. Once sufficient discussions have taken place under the RFC PR, the Maintainers will call a vote to decide whether the new SIG/project should be added.
```

winkyao commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,117 @@
+| Role | Responsibilities | Requirements | Defined by |
+| -----| ---------------- | ------------ | ------- |
+Contributor | contribute | initial contribution (PR, issue reporting, answering questions, etc.) | `Contributor List`|
+| Active Contributor | active contributor in the community | sponsored by 2 Reviewers. Multiple contributions (8+) to the project. | `SIG member`  of the SIG |
```
"sponsored by 2 Reviewers. Multiple contributions (8+) to the project." -> sponsored by 2 Reviewers. Multiple contributions to the project.

I think we should remove the constraints of digital metrics.

winkyao commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,117 @@
+### Proposal process
 
-### Organization
+We use a [Request for Comments (RFC) process for any substantial changes to TiKV. This process involves an upfront design that will provide increased visibility to the community. If you're considering a PR that will bring in a new feature that may affect how TiKV is implemented, or may be a breaking change, then you should start with a RFC. We've got the process documented in [RFC repository](htt ...
```
```suggestion
We use a Request for Comments (RFC) process for any substantial changes to TiKV. This process involves an upfront design that will provide increased visibility to the community. If you're considering a PR that will bring in a new feature that may affect how TiKV is implemented, or may be a breaking change, then you should start with a RFC. We've got the process documented in [RFC repository](https://github.com/tikv/rfcs) and have a [template](https://github.com/tikv/rfcs/blob/master/template.md) for you to get started.
```

dcalvin commented:
@nrc @winkyao @BusyJay PTAL again.  I know that it is ideal to base all on consensus, but I think voting from maintainers is also necessary and efficient for the listed scenarios.

nrc commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,106 @@
+
+# TiKV Governance
+
+ This document describes the governance rules of the TiKV project (organization). It is meant to be followed by all the repositories under the TiKV organization.
```
```suggestion
 This document describes the governance rules of the TiKV project (organization). It is meant to be followed by all projects in the TiKV organization.
```

Sorry, I know I already requested a change here

nrc commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,106 @@
+
+While maintainership indicates a valued member of the community who has demonstrated a healthy respect for the project’s aims and objectives, their work must still be reviewed by the community before being accepted into an official release.
+
+A Maintainer is not allowed to merge their change without approval from another person. However, they are allowed to sidestep this rule under justifiable circumstances that was approved through other means than the standard process.
```
```suggestion
A Maintainer is not allowed to merge their change without approval from another person. However, they are allowed to sidestep this rule under justifiable circumstances that was approved through other means than the standard process.
```
```suggestion
A Maintainer is not allowed to merge their changes without approval from another reviewer. However, they are allowed to sidestep this rule under exceptional circumstances if the changes are approved through other means than the standard process.
```

nrc commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,106 @@
+- Demonstrate a deep and comprehensive understanding of TiKV's codebase, technical goals, and directions
+- Handle complex problems in the code implementation process
+
+A new Maintainer must be self-nominated or nominated by an existing Maintainer by updating [TiKV Maintainers](https://github.com/tikv/tikv/blob/master/MAINTAINERS.md#the-tikv-maintainers). This triggers a voting process that requires supermajority votes from the current Maintainers.
```
I think allowing self-nomination is a mistake - if a candidate can't find a single maintainer to nominate them, then it's unlikely they'll be approved (for a candidate there is not much difference from self-nominating and asking a maintainer to nominate you).

dcalvin replied:
Good point.

nrc commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,106 @@
+## Decision making and voting
 
-Organizer Committee:  The Organizer Committee consists of organizers in charge of event or activity operations; in charge of the execution of PMC’s strategies and decisions. Specifically, the members are the leaders of user groups or community event organizers across regions
+Ideally, all project decisions are resolved by consensus via a PR or GitHub issue. Any of the day-to-day project maintenance can be done by a [lazy consensus](http://communitymgt.wikia.com/wiki/Lazy_consensus) model. Cross-SIG or community-level decisions must be brought to broader awareness via community meetings, slack channels, etc.
```
Is an RFC enough? I think it should be. There should probably be an async way to communicate this sort of thing, e.g. via the monthly newsletter (perhaps that is too low frequency?)

dcalvin replied:
Yeah  monthly community meeting and our monthly newsletter, plus slack should be enough for async communication

nrc replied:
Community meeting and Slack are both synchronous communication (because somebody in a different timezone might miss things). Monthly newsletter should be enough for async, we can experiment.

dcalvin commented on GOVERNANCE.md:
```diff
@@ -1,45 +1,102 @@
+
+- Enable and promote successful adoptions on the user or ecosystem end
+- Collaborate well
+- Demonstrate a deep and comprehensive understanding of TiKV's architecture, technical goals, and directions
```
I make these more general to make them include both maintainers from committers and eco-system.

sunxiaoguang commented:
LGTM
